runtime/doc/msgpack_rpc.txt

```diff
@@ -242,4 +242,198 @@ Even for statically compiled clients it is good practice to avoid hardcoding
 the type codes, because a client may be built against one Nvim version but
 connect to another with different type codes.
 
+==============================================================================
+7. Live Updates				      *live-updates* *rpc-live-updates*
+
+A dedicated API has been created to allow co-processes to be notified in
+real-time when the user changes a buffer in any way. It is difficult and
+error-prone to try and do this with autocommands such as |TextChanged|.
+
+							*live-updates-enabling*
+Setting Up~
+
+If your API client is a standalone co-process, it can use the
+`"nvim_buf_live_updates"`API method to activate Live Update events for a
+specific buffer. For example, in python >
+
+	import sys, neovim
+	nvim = neovim.attach('stdio')
+	bufnr = sys.argv[1]
+	nvim.buffers[bufnr].live_updates(True, True)
+
+After the `"nvim_buf_live_updates"` method is called, neovim will send a
+series of notifications containing the entire buffer's contents and any
+subsequent changes. The buffer's contents are sent via notifications because
+if you were to use the other API methods to retrieve the buffer contents, the
+buffer could be changed again before you turn on live updates. This can cause
+a delay if your plugin activates live updates for a very large buffer, but it
+is the the most efficient way to maintain a copy of the entire buffer's
+contents inside your plugin.
+
+To disable the initial sending of the full buffer, set the second argument of
+"nvim_buf_live_updates" to false. The resulting notification will be the same
+as shown in |LiveUpdateStart|, but {linedata} will be an empty list.
+
+						      *live-updates-disabling*
+Turning Off~
+
+You can use `"nvim_buf_live_updates"` with an argument of `False` to turn off
+notifications. One final notification will be sent to indicate that live
+updates are no longer active for the specified buffer. Alternatively, you can
+just close the channel.
+
+						    *live-updates-limitations*
+Limitations~
+
+Note that any of the following actions will also turn off live updates because
+the buffer contents are unloaded from memory:
+
+  - Closing all a buffer's windows (unless 'hidden' is enabled).
+  - Using |:edit| to reload the buffer
+  - reloading the buffer after it is changed from outside neovim.
+
+							  *live-updates-events*
+Handling Events~
+
+The co-process will start receiving the notification events which will be
+equivilent to the following |rpcnotify()| calls:
+
+1. rpcnotify({channel}, "LiveUpdateStart",		      *LiveUpdateStart*
```

Comment:
`rpcnotfy` shouldn't be used here, the fundamental concept is msgpack notifications (and what's relevant for the reader is how they are represented at the receiving end, which we don't control). Just mention name + `[argument, list]`

Reply:
Like [this](https://github.com/neovim/neovim/blob/master/runtime/doc/api.txt#L128)?

Reply:
Possibly, but I would like to make events visually distinct from methods in some way, maybe by using `[ ]` argument list.